.github/workflows/pr-title-spell-check.yml

```
@@ -0,0 +1,27 @@
+name: PR Title Spell Check
+
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  typos:
+    name: Spell check PR title
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

```suggestion
        uses: actions/checkout@v4
```
This updates the checkout to the latest version. More info here: https://github.com/actions/checkout

---

This can be done together for all the other workflows